packages/odyssey-react-mui/src/web-component/renderReactInWebComponent.ts

```diff
@@ -100,13 +101,24 @@ export type RenderReactInWebComponentProps = {
    * You will need to have rendered `<slot>` elements in your React component or `children` won't show up.
    */
   webComponentChildren?: HTMLElement | HTMLElement[];
-  /**
-   * You React app renders in the web component, but the web component needs to be rendered in the document.
-   *
-   * This is the element the web component is rendered into.
-   */
-  webComponentRootElement: HTMLElement;
-};
+} & (
+  | {
+      /**
+       * The React app renders in the web component, but the web component needs to be rendered in the document.
+       *
+       * This is the element the web component is rendered into.
+       */
+      webComponentParentElement: HTMLElement;
+      webComponentRootElement?: never;
+    }
+  | {
+      webComponentParentElement?: HTMLElement;
+      /**
+       * @deprecated Use `webComponentParentElement` instead.
```

Comment:
nit Can we just remove `webComponentRootElement` altogether in this PR or is it a lot of work to do?

Reply:
Nope. People are using this, and Odyssey doesn't do version bumps. This is part of core Odyssey, not UI Shell specifically, so I don't wanna break people.